node/src/components/linear_chain_sync/operations.rs

```diff
@@ -101,24 +107,83 @@ where
 {
     let block_header =
         fetch_retry_forever::<BlockHeader, REv, I>(effect_builder, block_hash).await?;
-    if !effect_builder
+    effect_builder
         .put_block_header_to_storage(block_header.clone())
-        .await
+        .await;
+    Ok(block_header)
+}
+
+fn validate_finality_signatures(
+    block_header: &BlockHeader,
+    trusted_validator_weights: &BTreeMap<PublicKey, U512>,
+    finality_threshold_fraction: Ratio<u64>,
+    block_signatures: &BlockSignatures,
+) -> Result<(), FinalitySignatureError> {
```

Comment:
@afck Please review!  I am going to be testing finality signatures from here on out... this passes my unit test with multiple nodes with random weights.